sbi/neural_nets/embedding_nets.py

```diff
@@ -266,19 +271,31 @@ def forward(self, x: Tensor) -> Tensor:
         Returns:
             Network output (batch_size, output_dim).
         """
-        batch, permutation_dim, _ = x.shape
 
-        iid_embeddings = self.trial_net(x.view(batch * permutation_dim, -1)).view(
-            batch, permutation_dim, -1
+        # Get number of trials from non-nan entries
+        trial_counts = x.shape[self.aggregation_dim] - torch.isnan(x[:, :, 0]).sum(
+            dim=1, keepdim=True
         )
 
-        if self.combining_operation == "mean":
-            e = iid_embeddings.mean(dim=1)
-        elif self.combining_operation == "sum":
-            e = iid_embeddings.sum(dim=1)
+        # get nan entries
+        is_nan = torch.isnan(x)
+        # apply trial net with nan entries replaced with 0
+        x = torch.nan_to_num(x, nan=0.0)
+        trial_embeddings = self.trial_net(x)
+        # replace previous nan entries with zeros
+        trial_embeddings = trial_embeddings * (~is_nan.all(-1, keepdim=True)).float()
```

Comment:
what if the user uses an `aggregation_fn` that does not map 0 to 0? Then these elements will have an influence although they should not have.

A simple fix would be to not allow a custom `aggregation_fn` (which I would be perfectly fine with)

Reply:
Or if the user simply passes `torch.mean` it will also fail. I fear that this will be a source of error for many users that is very hard to debug

Reply:
just saw that manuel also mentioned this. It's very sneaky imo, I would really prefer to remove the option of an `aggregation_fn` or, alternatively, force it to operate on a list of trial embeddings which are of different lenghts.

Reply:
I agree, it makes things a lot easier if we don't allow for custom aggregation functions. I changed it such that we only have `mean` or `sum`.